code/datums/gamemodes/crash.dm

```diff
@@ -0,0 +1,233 @@
+	var/planet_nuked = FALSE
+
+	var/shuttle_id = "tgs_canterbury"
+	var/obj/docking_port/mobile/crashmode/shuttle
+
+
+/datum/game_mode/crash/New()
+	. = ..()
+	xenomorphs = list()
+
+
+/datum/game_mode/crash/can_start()
+	. = ..()
+	// Check if enough players have signed up for xeno & queen roles.
+	var/ruler = initialize_xeno_leader()
+	var/xenos = initialize_xenomorphs()
+	if(!ruler && !xenos) // we need at least 1
+		return FALSE
+
+	return TRUE
+
+
+/datum/game_mode/crash/pre_setup()
+	. = ..()
+	// Spawn the ship
+	if(!SSmapping.shuttle_templates[shuttle_id])
+		CRASH("Shuttle [shuttle_id] wasn't found and can't be loaded")
+		return FALSE
+
+	var/datum/map_template/shuttle/S = SSmapping.shuttle_templates[shuttle_id]
+	var/obj/docking_port/stationary/L = SSshuttle.getDock("canterbury_loadingdock")
+	shuttle = SSshuttle.action_load(S, L)
+
+	// Reset all spawnpoints after spawning the ship
+	GLOB.marine_spawns_by_job = shuttle.marine_spawns_by_job
+	GLOB.jobspawn_overrides = list()
+	GLOB.latejoin = shuttle.spawnpoints
+	GLOB.latejoin_cryo = shuttle.spawnpoints
+	GLOB.latejoin_gateway = shuttle.spawnpoints
+
+	// Create xenos
+	var/number_of_xenos = length(xenomorphs)
+	var/datum/hive_status/normal/HN = GLOB.hive_datums[XENO_HIVE_NORMAL]
+	for(var/i in xenomorphs)
+		var/datum/mind/M = i
+		if(M.assigned_role == ROLE_XENO_QUEEN)
+			transform_ruler(M, number_of_xenos > HN.xenos_per_queen)
+		else
+			transform_xeno(M)
+
+
+/datum/game_mode/crash/post_setup()
+	. = ..()
+
+	// Reset all spawnpoints after spawning the ship
+	GLOB.marine_spawns_by_job = shuttle.marine_spawns_by_job
+	GLOB.jobspawn_overrides = list()
+	GLOB.latejoin = shuttle.spawnpoints
+	GLOB.latejoin_cryo = shuttle.spawnpoints
+	GLOB.latejoin_gateway = shuttle.spawnpoints
+
+	var/list/validdocks = list()
+	for(var/obj/docking_port/stationary/S in SSshuttle.stationary)
+		if(!shuttle.check_dock(S, silent=TRUE))
+			continue
+		validdocks += S.name
+
+	var/dock = pick(validdocks)
+
+	var/obj/docking_port/stationary/target
+	for(var/obj/docking_port/stationary/S in SSshuttle.stationary)
+		if(S.name == dock)
+			target = S
+			break
+
+	if(!target)
+		CRASH("Unable to get a valid shuttle target!")
+		return
+
+	addtimer(CALLBACK(src, .proc/crash_shuttle, target), 9 MINUTES) // TODO: REMOVE ADMIN TIMING
+	// addtimer(CALLBACK(src, .proc/crash_shuttle, target), rand(9 MINUTES, 11 MINUTES)) // TODO: FIX timing here
+
+
+/datum/game_mode/crash/announce()
+	to_chat(world, "<span class='round_header'>The current map is - [SSmapping.configs[GROUND_MAP].map_name]!</span>")
+	priority_announce("Scheduled for landing in T-10 Minutes. Prepare for landing. Known hostiles near LZ. Detonation Protocol Active, planet disposable. Marines disposable.", type = ANNOUNCEMENT_PRIORITY) // TODO: Better text.
+	playsound(shuttle, 'sound/machines/warning-buzzer.ogg', 75, 0, 30)
+	
+
+/datum/game_mode/crash/proc/add_larva()
+	var/datum/hive_status/normal/HS = GLOB.hive_datums[XENO_HIVE_NORMAL]
```

Comment:
Add a condition here to produce xeno spawns depending on number of marines.
One larva every 5 minutes without a cap may quickly cause infinite xenos.
Perhaps a check every 10 minutes:
* If the xenos+larvas are less than 1 per 4 marines, give them multiple larvas and message the hive informing them.
* If there are more, give them one larva.
* If there are same number of xenos (or more) than marines, give them nothing.

Reply:
Something like:
```
if(!numxenos)
	if(world.time < CRASH_MINIMUM_TIME + SSticker.round_start_time) //Xenos keep respawning for like an hour or so.
		spawn_xenos()
		return
	else
		end_game()
		return //No more xenos.
var/marines_per_xeno = nummarines / numxenos
switch(marines_per_xeno)
	if(0)
		end_game()
		return //No more marines.
	if(0 to 1) //Same number of marines and xenos, or more xenos.
		return
	if(1 to 4)
		larvas++
	else
		larvas += CLAMP(round(marines_per_xeno * 0.3), 1, nummarines - numxenos)
```